src/handlers/list.ts

```diff
@@ -1,10 +1,28 @@
+import { readFile } from "node:fs/promises";
+import { Task } from "../models/Task";
+
 export namespace ListHandler {
   export type Args = {
     all: boolean;
   };
 
+  const taskData = "data/db.json";
+
   export const handle = async (args: Args): Promise<void> => {
-    // TODO
-    throw new Error("Not yet implemented");
+    const data = await readFile(taskData);
+    const stringContent = data.toString();
+    const taskObjects: Array<Task> = JSON.parse(stringContent);
+
+    const taskStrings: Array<string> = taskObjects.map((task) => {
+      if (task.done) {
+        return `[x] ${task.description}`;
+      } else {
+        return `[ ] ${task.description}`;
+      }
```

Comment:
We could simplify this to something like:

```suggestion
      return `[${task.done ? 'x' : ' '}] ${task.description}`;
```

Reply:
That's amazing! But I would like to keep the old version so that I can understand what is going on here.. I will keep improving so that I will be able to understand codes like what you suggested! Thank you!